### PR TITLE
Fixes #22

### LINKED ISSUE #22: Update metadata
Just noticed there's a `create` method and a `get_metadata` method using the same endpoint, but no `update` method. Here's some relevant info from Socrata on the `/api/views.json` endpoint:

> `PUT` to `/api/views/unique_id.json` with a subset of what you want to update.  For instance, if you want to add tags to the previously created dataset (currently this is the only way to add tags), you can simply `PUT` to the endpoint with:
```json
{
  "tags": ["these", "words", "are", "tags"]
}
```
> You'll probably also want to make this dataset public and publish it, so you'll want to perform the two following operations as well:
`PUT /views/unique_id?method=setPermission&value=public.read`
`POST /api/views/unique_id/publication.json`

> One important thing to note about updates is that if you want to update something within the "custom metadata", you will have to copy the entire metadata object since it does a replace.

Note that I've heard over the year(s) that the views endpoint is/will be deprecated, but until there's a clear alternative it probably makes sense to continue to use it.

Anyway, I may get a chance to work on this, but wanted to post the details here in case anyone else is interested.

### PR DESCRIPTION
**Pull request by xmunoz**
Add `update_metdata` method, test and documentation.

### CODE REVIEW

**Comment by xmunoz** [comment]
@chrismetcalf @timwis take a look at this. Also, I've added you both as collaborators, so that either of you in the future can merge code or close issues.

File: README [hunk 1]
```diff
@@ -101,6 +104,21 @@ Retrieve the metadata associated with a particular dataset.
     >>> client.get_metadata("nimj-3ivp")
     {"newBackend": false, "licenseId": "CC0_10", "publicationDate": 1436655117, "viewLastModified": 1451289003, "owner": {"roleName": "administrator", "rights": [], "displayName": "Brett", "id": "cdqe-xcn5", "screenName": "Brett"}, "query": {}, "id": "songs", "createdAt": 1398014181, "category": "Public Safety", "publicationAppendEnabled": true, "publicationStage": "published", "rowsUpdatedBy": "cdqe-xcn5", "publicationGroup": 1552205, "displayType": "table", "state": "normal", "attributionLink": "http://foo.bar.com", "tableId": 3523378, "columns": [], "metadata": {"rdfSubject": "0", "renderTypeConfig": {"visible": {"table": true}}, "availableDisplayTypes": ["table", "fatrow", "page"], "attachments": ... }}
 
+update\_metadata(dataset\_identifier, update\_fields, content\_type="json")
```
Comment (chrismetcalf) [inline README]: I might include a statement like "To be safe, pass a dictionary containing only the keys you wish to update."

**Comment by xmunoz** [comment]
I tested a couple of keys in the metadata, and for most it changed them. For some others though, it doesn't. An example of this is "category". If the user tried to set a category to one that isn't in the prefined list in the UI, then category will simply be undefined. This is a bit confusing, because the api returns a 200 response, but I guess that is a case that you guys will handle when redesigning this endpoint.

In summary, I'm just returning whatever the server returns, for better or worse.

**Comment by timwis** [comment]
This is great! Thanks so much for working on this. I love that the actual code for the method is only 2 lines :P

Regarding the category, I noticed that with `create` as well, and noted in the docs that it must exist in `/admin/metadata`.
